examples/aggregator/module/rust/src/router.rs

```diff
@@ -0,0 +1,137 @@
+///
+/// This node never looks at the contents of the invocation messages, only at the labels of its
+/// channels, and therefore keeps a public confidentiality label, which also allows it to create
+/// further nodes and channels, with more specific labels.
+#[derive(Default)]
+pub struct Router {
+    /// Invocation sender channel half for Handler Node.
+    handler_command_sender: Sender<oak::grpc::Invocation>,
+    /// Confidentiality label corresponding to Aggregator WebAssembly module SHA-256 hash.
+    aggregator_hash_label: Label,
+}
+
+impl oak::CommandHandler for Router {
+    type Command = oak::grpc::Invocation;
+
+    fn handle_command(&mut self, command: oak::grpc::Invocation) -> anyhow::Result<()> {
+        // The router node has a public confidentiality label, and therefore cannot read the
+        // contents of the request of the invocation (unless it happens to be public as well), but
+        // it can always inspect its label.
+        match (&command.receiver, &command.sender) {
+            (Some(receiver), Some(sender)) => {
+                let label = receiver.label()?;
+                let grpc_response_writer = oak::grpc::ChannelResponseWriter::new(sender.clone());
+                // Check if request label is valid in the context of Aggregator application.
+                if label != self.aggregator_hash_label {
+                    grpc_response_writer
+                        .close(Err(oak::grpc::build_status(
+                            rpc::Code::InvalidArgument,
+                            "Invalid request label",
+                        )))
+                        .context("Couldn't send error response back")?;
+                    anyhow::bail!("Invalid request label: {:?}", label);
+                }
+
+                // Route gRPC invocations to Handler Node.
+                self.handler_command_sender
+                    .send(&command)
+                    .context("Couldn't send invocation to worker node")
+            }
+            _ => {
+                anyhow::bail!("Received malformed gRPC invocation");
+            }
+        }
+    }
+}
+
+impl oak::WithInit for Router {
+    type Init = ConfigMap;
+
+    fn create(init: Self::Init) -> Self {
+        let config: Config =
+            toml::from_slice(&init.items.get("config").expect("Couldn't find config"))
+                .expect("Couldn't parse TOML config file");
+
+        // TODO(#1674): Uncomment and use correct confidentiality labels.
+        // let label = Label {
+        //     confidentiality_tags: vec![Tag {
+        //         tag: Some(tag::Tag::TlsEndpointTag(TlsEndpointTag {
+        //             certificate_subject_alternative_name: "google.com".to_string(),
+        //         })),
+        //     }],
+        //     integrity_tags: vec![],
+        // };
+        let grpc_client_invocation_sender = oak::grpc::client::init(&config.backend_server_address)
+            .expect("Couldn't create gRPC client");
```

Comment:
Handler is also created inside `Router`. Will probably create all nodes in the Main node.
But this will leave `Router` only the check of incoming labels (which later on, I think, should be a part of SDK).